pkg/operator/certrotation/signer.go

```diff
@@ -212,9 +216,9 @@ func setSigningCertKeyPairSecretAndTLSAnnotations(signingCertKeyPairSecret *core
 }
 
 // setSigningCertKeyPairSecret creates a new signing cert/key pair and sets them in the secret
-func setSigningCertKeyPairSecret(signingCertKeyPairSecret *corev1.Secret, validity time.Duration) (*crypto.TLSCertificateConfig, error) {
-	signerName := fmt.Sprintf("%s_%s@%d", signingCertKeyPairSecret.Namespace, signingCertKeyPairSecret.Name, time.Now().Unix())
-	ca, err := crypto.MakeSelfSignedCAConfigForDuration(signerName, validity)
+func setSigningCertKeyPairSecret(signingCertKeyPairSecret *corev1.Secret, now time.Time, validity time.Duration) (*crypto.TLSCertificateConfig, error) {
+	signerName := fmt.Sprintf("%s_%s@%d", signingCertKeyPairSecret.Namespace, signingCertKeyPairSecret.Name, now.Unix())
+	ca, err := crypto.UnsafeMakeSelfSignedCAConfigForDurationAtTime(signerName, func() time.Time { return now }, validity)
```

Comment:
Unsafe ?!

i'd prefer changing `MakeSelfSignedCAConfigForDuration` to accept `clock.Clock`

Reply:
I don't know why the method is called like that, makes no sense TBH.